packages/next/src/server/lib/router-server.ts

```diff
@@ -585,13 +585,20 @@ export async function initialize(opts: {
       })
 
       if (opts.dev && developmentBundler && req.url) {
-        const isHMRRequest = req.url.includes('/_next/webpack-hmr')
+        const { basePath } = config
+
+        const isHMRRequestForHostApp =
+          !basePath && req.url.startsWith('/_next/webpack-hmr')
```

Comment:
What about a single application that specifies a `basePath`? A request to that application still seems like the "host app" as multi-zone is not the only use-case for the `basePath` property.

Reply:
@ztanner I decided to improve verbiage here to make it clear, so now I updated it from `isHMRRequestForHostApp` to `isHMRRequestWithoutBasePath` and `isHMRRequestForGuestApp` to `isHMRRequestWithBasePath`.
Now a single application that have basePath will be handled by `isHMRRequestWithBasePath` condition 🚀